### PR TITLE
Add water calc methods

### LINKED ISSUE #23: Add Methods for Calculating Water Demand
image

#### Description:
Creating Demand Side Calculation for Water (Plumbing fixtures, cooling towers and irrigation)

### PR DESCRIPTION
**Pull request by kayleighhoude**
### Issues addressed by this PR

Closes #23 

### Additional comments

Plumbing Fixture Water Usage Method

### CODE REVIEW

File: ClimateEmergency_Engine/Compute/Water/PlumbingFixtureWaterDemandDaily.cs [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+        /***************************************************/
+        /****   Public Methods                          ****/
+        /***************************************************/
+
+        [Description("Calculates the plumbing fixture water demand per day using the plumbing fixture flows and the building occupancy.")]
+        [Input("BuildingPlumbingFixtureSelection", "The BHoM Object which contains the selected building plumbing fixtures. This method requires the object's flow per fixture to be set.")]
+        [Input("BuildingOccupancybyGender", "The BHoM Object which contains the building's occupancy by gender. This method requires the object's occupancy percentages to be set.")]
+        [Input("FixtureUsageDataSet", "The dataset or custom object which contains the residential or commercial building based daily usage of plumbing fixtures.")]
+        [Output("LitersofWaterPerDay", "The numbers of liters of water used on a daily basis by plumbing fixtures.")]
+
```
Comment (FraserGreenroyd) [inline ClimateEmergency_Engine/Compute/Water/PlumbingFixtureWaterDemandDaily.cs]: Remove this blank line

File: ClimateEmergency_Engine/Compute/Water/PlumbingFixtureWaterDemandDaily.cs [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+        public static double PlumbingFixtureWaterDemandPerDay(BuildingOccupancyByGender BuildingOccupancybyGender, BuildingPlumbingFixtureSelection BuildingPlumbingFixtureSelection, CustomObject FixtureUsageDataSet)
```
Comment (FraserGreenroyd) [inline ClimateEmergency_Engine/Compute/Water/PlumbingFixtureWaterDemandDaily.cs]: Input parameter variable names should start with a lowercase letter

File: ClimateEmergency_Engine/Compute/Water/PlumbingFixtureWaterDemandDaily.cs [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+            {
+                kitchenfaucetnumberofusesmale = (double)FixtureUsageDataSet.CustomData["KitchenFaucetNumberOfUsesMale"];
+            }
+            else
+            {
+                BH.Engine.Reflection.Compute.RecordError("The BHoMObject must have a valid KitchenFaucetNumberOfUsesMale stored in CustomData under a 'kitchenfaucetNumberOfUsesMale' key.");
+                return 0;
+            }
+
+            double kitchenfaucetnumberofusesfemale = 0.0;
```
Comment (FraserGreenroyd) [inline ClimateEmergency_Engine/Compute/Water/PlumbingFixtureWaterDemandDaily.cs]: Can we camel case this variable `kitchenFaucetNumberOfUsesFemale`

File: ClimateEmergency_Engine/Compute/Water/PlumbingFixtureWaterDemandDaily.cs [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+            {
+                kitchenfaucetnumberofusesfemale = (double)FixtureUsageDataSet.CustomData["KitchenFaucetNumberOfUsesFemale"];
+            }
+            else
+            {
+                BH.Engine.Reflection.Compute.RecordError("The BHoMObject must have a valid KitchenFaucetNumberOfUsesFemale stored in CustomData under a 'kitchenfaucetNumberOfUsesMale' key.");
+                return 0;
+            }
+
+            double kitchenfaucetnumberofusesgenderneutral = 0.0;
```
Comment (FraserGreenroyd) [inline ClimateEmergency_Engine/Compute/Water/PlumbingFixtureWaterDemandDaily.cs]: Camel case please

File: ClimateEmergency_Engine/Compute/Water/PlumbingFixtureWaterDemandDaily.cs [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+            {
+                kitchenfaucetnumberofusesgenderneutral = (double)FixtureUsageDataSet.CustomData["KitchenFaucetNumberOfUsesGenderNeutral"];
+            }
+            else
+            {
+                BH.Engine.Reflection.Compute.RecordError("The BHoMObject must have a valid KitchenFaucetNumberOfUsesGenderNeutral stored in CustomData under a 'kitchenfaucetNumberOfUsesMale' key.");
+                return 0;
+            }
+
+            double lavatorynumberofusesmale = 0.0;
```
Comment (FraserGreenroyd) [inline ClimateEmergency_Engine/Compute/Water/PlumbingFixtureWaterDemandDaily.cs]: Camel case please

File: ClimateEmergency_Engine/Compute/Water/PlumbingFixtureWaterDemandDaily.cs [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+            {
+                lavatorynumberofusesmale = (double)FixtureUsageDataSet.CustomData["LavatoryNumberOfUsesMale"];
+            }
+            else
+            {
+                BH.Engine.Reflection.Compute.RecordError("The BHoMObject must have a valid LavatoryNumberOfUsesMale stored in CustomData under a 'lavatoryNumberOfUsesMale' key.");
+                return 0;
+            }
+
+            double lavatorynumberofusesfemale = 0.0;
```
Comment (FraserGreenroyd) [inline ClimateEmergency_Engine/Compute/Water/PlumbingFixtureWaterDemandDaily.cs]: Camel case please

File: ClimateEmergency_Engine/Compute/Water/PlumbingFixtureWaterDemandDaily.cs [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+            {
+                lavatorynumberofusesfemale = (double)FixtureUsageDataSet.CustomData["LavatoryNumberOfUsesFemale"];
+            }
+            else
+            {
+                BH.Engine.Reflection.Compute.RecordError("The BHoMObject must have a valid LavatoryNumberOfUsesFemale stored in CustomData under a 'lavatoryNumberOfUsesMale' key.");
+                return 0;
+            }
+
+            double lavatorynumberofusesgenderneutral = 0.0;
```
Comment (FraserGreenroyd) [inline ClimateEmergency_Engine/Compute/Water/PlumbingFixtureWaterDemandDaily.cs]: Camel case please

File: ClimateEmergency_Engine/Compute/Water/PlumbingFixtureWaterDemandDaily.cs [hunk 8]
```diff
@@ -0,0 +1,232 @@
...
+            {
+                lavatorynumberofusesgenderneutral = (double)FixtureUsageDataSet.CustomData["LavatoryNumberOfUsesGenderNeutral"];
+            }
+            else
+            {
+                BH.Engine.Reflection.Compute.RecordError("The BHoMObject must have a valid LavatoryNumberOfUsesGenderNeutral stored in CustomData under a 'lavatoryNumberOfUsesMale' key.");
+                return 0;
+            }
+
+            double urinalnumberofusesmale = 0.0;
```
Comment (FraserGreenroyd) [inline ClimateEmergency_Engine/Compute/Water/PlumbingFixtureWaterDemandDaily.cs]: Camel case please

File: ClimateEmergency_Engine/Compute/Water/PlumbingFixtureWaterDemandDaily.cs [hunk 9]
```diff
@@ -0,0 +1,232 @@
...
+            {
+                urinalnumberofusesmale = (double)FixtureUsageDataSet.CustomData["UrinalNumberOfUsesMale"];
+            }
+            else
+            {
+                BH.Engine.Reflection.Compute.RecordError("The BHoMObject must have a valid UrinalNumberOfUsesMale stored in CustomData under a 'urinalNumberOfUsesMale' key.");
+                return 0;
+            }
+
+            double totaldailytoiletflow = (numbermales * toiletnumberofusesmale * toiletflow) + (numberfemales * toiletnumberofusesfemale * toiletflow) + (numbergenderneutral * toiletnumberofusesgenderneutral * toiletflow);
```
Comment (FraserGreenroyd) [inline ClimateEmergency_Engine/Compute/Water/PlumbingFixtureWaterDemandDaily.cs]: Camel case please

File: ClimateEmergency_Engine/Create/Water/BuildingOccupancyByGender.cs [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Returns the building occupancies by gender")]
+        [Input("buildingOccupancy", "Default 0")]
+        [Input("femalePercentage", "Default 0.5")]
+        [Input("malePercentage", "Default 0.5")]
+        [Input("genderNeutralPercentage", "Default 0.0")]
+        [Output("building occupancy by gender object")]
+
```
Comment (FraserGreenroyd) [inline ClimateEmergency_Engine/Create/Water/BuildingOccupancyByGender.cs]: Delete blank line

File: ClimateEmergency_Engine/Create/Water/BuildingPlumbingFixtureSelection.cs [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+        /***************************************************/
+
+        [Description("Returns the building plumbing fixture selection")]
+        [Input("toiletFlow", "Default 0.0")]
+        [Input("urinalFlow", "Default 0.0")]
+        [Input("lavatoryFlow", "Default 0.0")]
+        [Input("showerFlow", "Default 0.0")]
+        [Input("kitchenFaucetFlow", "Default 0.0")]
+        [Output("building plumbing fixture selection object")]
+
```
Comment (FraserGreenroyd) [inline ClimateEmergency_Engine/Create/Water/BuildingPlumbingFixtureSelection.cs]: Delete blank line

File: ClimateEmergency_oM/Water/BuildingOccupancyByGender.cs [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.ComponentModel;
+using BH.oM.Base;
+
+namespace BH.oM.ClimateEmergency.Water
+{
+    public class BuildingOccupancyByGender
+    {
+
```
Comment (FraserGreenroyd) [inline ClimateEmergency_oM/Water/BuildingOccupancyByGender.cs]: Delete blank line

File: ClimateEmergency_oM/Water/BuildingPlumbingFixtureSelection.cs [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.ComponentModel;
+using BH.oM.Base;
+
+namespace BH.oM.ClimateEmergency.Water
+{
+    public class BuildingPlumbingFixtureSelection
+    {
+
```
Comment (FraserGreenroyd) [inline ClimateEmergency_oM/Water/BuildingPlumbingFixtureSelection.cs]: Delete blank line

File: ClimateEmergency_Engine/Compute/Water/PlumbingFixtureWaterDemandDaily.cs [hunk 14]
```diff
@@ -0,0 +1,232 @@
...
+    public static partial class Compute
+    {
+        /***************************************************/
+        /****   Public Methods                          ****/
+        /***************************************************/
+
+        [Description("Calculates the plumbing fixture water demand per day using the plumbing fixture flows and the building occupancy.")]
+        [Input("BuildingPlumbingFixtureSelection", "The BHoM Object which contains the selected building plumbing fixtures. This method requires the object's flow per fixture to be set.")]
+        [Input("BuildingOccupancybyGender", "The BHoM Object which contains the building's occupancy by gender. This method requires the object's occupancy percentages to be set.")]
+        [Input("FixtureUsageDataSet", "The dataset or custom object which contains the residential or commercial building based daily usage of plumbing fixtures.")]
```
Comment (michaelhoehn) [inline ClimateEmergency_Engine/Compute/Water/PlumbingFixtureWaterDemandDaily.cs]: Input lowercase

File: ClimateEmergency_Engine/Compute/Water/PlumbingFixtureWaterDemandDaily.cs [hunk 15]
```diff
@@ -0,0 +1,232 @@
...
+{
+    public static partial class Compute
+    {
+        /***************************************************/
+        /****   Public Methods                          ****/
+        /***************************************************/
+
+        [Description("Calculates the plumbing fixture water demand per day using the plumbing fixture flows and the building occupancy.")]
+        [Input("BuildingPlumbingFixtureSelection", "The BHoM Object which contains the selected building plumbing fixtures. This method requires the object's flow per fixture to be set.")]
+        [Input("BuildingOccupancybyGender", "The BHoM Object which contains the building's occupancy by gender. This method requires the object's occupancy percentages to be set.")]
```
Comment (michaelhoehn) [inline ClimateEmergency_Engine/Compute/Water/PlumbingFixtureWaterDemandDaily.cs]: Input lowercase

File: ClimateEmergency_Engine/Compute/Water/PlumbingFixtureWaterDemandDaily.cs [hunk 16]
```diff
@@ -0,0 +1,232 @@
...
+namespace BH.Engine.ClimateEmergency
+{
+    public static partial class Compute
+    {
+        /***************************************************/
+        /****   Public Methods                          ****/
+        /***************************************************/
+
+        [Description("Calculates the plumbing fixture water demand per day using the plumbing fixture flows and the building occupancy.")]
+        [Input("BuildingPlumbingFixtureSelection", "The BHoM Object which contains the selected building plumbing fixtures. This method requires the object's flow per fixture to be set.")]
```
Comment (michaelhoehn) [inline ClimateEmergency_Engine/Compute/Water/PlumbingFixtureWaterDemandDaily.cs]: Input lowercase - adjust variable names throughout

File: ClimateEmergency_Engine/Compute/Water/PlumbingFixtureWaterDemandDaily.cs [hunk 17]
```diff
@@ -0,0 +1,232 @@
...
+            double numbermales = (double)BuildingOccupancybyGender.BuildingOccupancy * BuildingOccupancybyGender.MalePercentage;
+            double numbergenderneutral = (double)BuildingOccupancybyGender.BuildingOccupancy * BuildingOccupancybyGender.GenderNeutralPercentage;
+
+            double toiletflow = BuildingPlumbingFixtureSelection.ToiletFlow;
+            double lavatoryflow = BuildingPlumbingFixtureSelection.LavatoryFlow;
+            double urinalflow = BuildingPlumbingFixtureSelection.UrinalFlow;
+            double showerflow = BuildingPlumbingFixtureSelection.ShowerFlow;
+            double kitchenfaucetflow = BuildingPlumbingFixtureSelection.KitchenFaucetFlow;
+
+            double toiletnumberofusesmale = 0.0;
```
Comment (michaelhoehn) [inline ClimateEmergency_Engine/Compute/Water/PlumbingFixtureWaterDemandDaily.cs]: Camelcase

**Comment by kristenbrozowski** [comment]
image

Please state source of fixture usage dataset (assuming from LEED)

**Comment by kristenbrozowski** [comment]
image

Correct the building plumbing fixture selection inputs from "ToiletFlow" to "ToiletVolumeperUse"

**Comment by FraserGreenroyd** [comment]
> image
> 
> Correct the building plumbing fixture selection inputs from "ToiletFlow" to "ToiletVolumeperUse"

I'd also recommend capitalising the `P` in `Per` on all outputs :smile:

**Comment by kristenbrozowski** [comment]
image

Remove the units on the output (they are m^3). Instead, place this in the description for the method.

File: ClimateEmergency_Engine/Compute/Water/PlumbingFixtureWaterDemandDaily.cs [hunk 18]
```diff
@@ -37,189 +37,187 @@ public static partial class Compute
...
-        [Input("BuildingPlumbingFixtureSelection", "The BHoM Object which contains the selected building plumbing fixtures. This method requires the object's flow per fixture to be set.")]
-        [Input("BuildingOccupancybyGender", "The BHoM Object which contains the building's occupancy by gender. This method requires the object's occupancy percentages to be set.")]
-        [Input("FixtureUsageDataSet", "The dataset or custom object which contains the residential or commercial building based daily usage of plumbing fixtures.")]
+        [Input("buildingPlumbingFixtureSelection", "The BHoM Object which contains the selected building plumbing fixtures. This method requires the object's flow per fixture to be set.")]
+        [Input("buildingOccupancybyGender", "The BHoM Object which contains the building's occupancy by gender. This method requires the object's occupancy percentages to be set.")]
+        [Input("fixtureUsageDataSet", "The dataset or custom object which contains the residential or commercial building based daily usage of plumbing fixtures.")]
         [Output("LitersofWaterPerDay", "The numbers of liters of water used on a daily basis by plumbing fixtures.")]
```
Comment (michaelhoehn) [inline ClimateEmergency_Engine/Compute/Water/PlumbingFixtureWaterDemandDaily.cs]: litersOfWaterPerDay

File: ClimateEmergency_Engine/Compute/Water/PlumbingFixtureWaterDemandDaily.cs [hunk 19]
```diff
@@ -37,189 +37,187 @@ public static partial class Compute
         /***************************************************/
 
         [Description("Calculates the plumbing fixture water demand per day using the plumbing fixture flows and the building occupancy.")]
-        [Input("BuildingPlumbingFixtureSelection", "The BHoM Object which contains the selected building plumbing fixtures. This method requires the object's flow per fixture to be set.")]
-        [Input("BuildingOccupancybyGender", "The BHoM Object which contains the building's occupancy by gender. This method requires the object's occupancy percentages to be set.")]
-        [Input("FixtureUsageDataSet", "The dataset or custom object which contains the residential or commercial building based daily usage of plumbing fixtures.")]
+        [Input("buildingPlumbingFixtureSelection", "The BHoM Object which contains the selected building plumbing fixtures. This method requires the object's flow per fixture to be set.")]
+        [Input("buildingOccupancybyGender", "The BHoM Object which contains the building's occupancy by gender. This method requires the object's occupancy percentages to be set.")]
```
Comment (michaelhoehn) [inline ClimateEmergency_Engine/Compute/Water/PlumbingFixtureWaterDemandDaily.cs]: buildingOccupancyByGender

File: ClimateEmergency_Engine/Compute/Water/PlumbingFixtureWaterDemandDaily.cs [hunk 20]
```diff
@@ -37,189 +37,187 @@ public static partial class Compute
...
+        [Input("fixtureUsageDataSet", "The dataset or custom object which contains the residential or commercial building based daily usage of plumbing fixtures.")]
         [Output("LitersofWaterPerDay", "The numbers of liters of water used on a daily basis by plumbing fixtures.")]
-
-        public static double PlumbingFixtureWaterDemandPerDay(BuildingOccupancyByGender BuildingOccupancybyGender, BuildingPlumbingFixtureSelection BuildingPlumbingFixtureSelection, CustomObject FixtureUsageDataSet)
+        public static double plumbingFixtureWaterDemandPerDay(BuildingOccupancyByGender buildingOccupancybyGender, BuildingPlumbingFixtureSelection buildingPlumbingFixtureSelection, CustomObject fixtureUsageDataSet)
         {
+            double numberFemales = (double)buildingOccupancybyGender.BuildingOccupancy * buildingOccupancybyGender.FemalePercentage;
```
Comment (michaelhoehn) [inline ClimateEmergency_Engine/Compute/Water/PlumbingFixtureWaterDemandDaily.cs]: buildingOccupancyByGender

**Comment by michaelhoehn** [comment]
All changes made and all checks passed. Merge requested @FraserGreenroyd

**Pull request merged**